Makefile

```diff
@@ -1,7 +1,8 @@
 all: amosbank
 
 amosbank: amosbank.cc
-	g++ -o amosbank amosbank.cc -lpng
+	g++ -o amosbank amosbank.cc  -lpng -lz
```

Comment:
It looks like this is indeed platform (or libpng version) dependent. The right thing to do here is using `$(libpng-config --libs)`.